core/util/instagram.py

```diff
@@ -0,0 +1,255 @@
+"""
+OWASP Maryam!
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import requests
+import json
+
+class main:
+    def __init__(self, q, limit=50, session_id=''):
+        """ instagram.com search
+            
+            q 		  : The query for search
+            limit	  : The number of details min 50 if exist
+            session_id: Your Instagram session id
+        """
+        
+        self.q = q
+        self.limit = int(limit/50) or 1
+        self.session_id = session_id
+        self.base_url = 'https://www.instagram.com/'
+        self.headers = {
+                	    'User-Agent': self.framework.rand_uagent().get,
+                        'Content-Type': 'application/x-www-form-urlencoded',
+                        }
+        if self.session_id:
+            self.headers['Cookie'] = f"sessionid={self.session_id};"
+
+        # request requrements
+        self.data = {}
+        self.posts_next_page = None
+        self.followers_next_page = None
+        self.following_next_page  = None
+        self.is_private = None
+
+        # output_data
+        self._userdata = {}
+        self._followers = []
+        self._following = []
+        self._post = []
+
+        self.session = requests.Session()
+
+    def run_crawl(self):
+        self.framework.verbose('[INSTAGRAM] Extracting Data From API')
+        # 1.1 account details find user on instagram 
+        req = self.get_user_account_info()
+        if req.status_code == 404 or self.data == {} :
+            # user not found - show similar account names
+            self.framework.error('Account not Found.')
+            similar_users = self.get_similar_users()
+            if similar_users:
+                self.framework.heading('showing similar users', 0)
+
+            for index,user in enumerate(similar_users, 1):
+                self.framework.alert(f"{index}. user : {user['username']} --> {self.base_url + user['username']}")
+            return
+
+        if self.is_private or req.status_code != 200 :
+            # return if account is private or request fails
+            if self.is_private : 
+                self.framework.error('Account is Private.')
+                if self.session_id == '':
+                    return
+
+            else :
+                self.framework.error(f"Request Fails, Url : {req.url}")
```

Comment:
Why? the [INSTAGRAM API] needs to be the first of all of the messages. I corrected it.